### PR TITLE
Added files for infinite 1-D chain band structure and decay rate calc…

### PR DESCRIPTION
**Pull request by taylorpatti**
…ulations.

### CODE REVIEW

File: src/collective_modes.jl [hunk 1]
```diff
@@ -0,0 +1,66 @@
+module collective_modes
+
+# For a particularly elegant derivation of collective frequency shifts and decay rates in 1D
+# atomic chains, see Asenjo-Garcia et al 10.1103/PhysRevX.7.031024.
+
+import LinearAlgebra
+import Polylogarithms
+
+pl = Polylogarithms
```
Comment (david-pl) [inline src/collective_modes.jl:9]: `const pl = Polylogarithms` just to be sure

File: src/collective_modes.jl [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+function Gamma_k_chain(k::Real, a::Real, polarization::Array{<:Number, 1})
+    if k > 2pi
+        return 0
+    end
+    polarization = LinearAlgebra.normalize(polarization)
+    x_par = polarization[1]^2
+    k0 = 2pi
+    g = 2pi/a
+    N = floor((k0 - k)/g)
+    sum = 0
```
Comment (david-pl) [inline src/collective_modes.jl:58]: `sum` is a Julia Base function, I'd rather not use it as a variable name

File: src/collective_modes.jl [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import Polylogarithms
+
+pl = Polylogarithms
+
+"""
+    Omega_k_chain(k, a, polarization)
+
+Collective frequency shift Omega_k of mode k for an infinite chain of atoms along x-axis.
+
+WLOG, this calculation scales natural atomic linewidth lambda0=1 and decay rate gamma0=1.
```
Comment (david-pl) [inline src/collective_modes.jl:16]: I'm a bit confused with the terminology. To me, the "natural atomic linewidth" is the decay rate. Are you referring to the transition wavelength as `lambda0`?

**Review by david-pl — COMMENTED** [review]
@taylorpatti Thanks for the work! Things look good, and the tests seem solid. I'd love to see an example showing a nice band structure plot, but that's probably for another time. I added some comments in the code, mainly stylistic stuff though.

One other thing you may want to consider is to add the `collective_modes` module to the list of exports, which would reduce the amount one has to type when calling the functions a bit.

Would you like to add 2D functionality here, or leave that for a separate PR?

**Comment by taylorpatti** [comment]
Hi @david-pl, 

Glad you like it! I can make those style adjustments - can you merge the pull request first? I'd like to set up a proper developer environment first and it will make it easier to make a new branch off the master.

I was planning to put the 2D functions in the same module. Although, I figured that I would do the finite system 1D functions, first. I'm fine with either order though, if there is a preference for the opposite.

**Comment by david-pl** [comment]
> Glad you like it! I can make those style adjustments - can you merge the pull request first? I'd like to set up a proper developer environment first and it will make it easier to make a new branch off the master.

I'm not sure I understand what you mean. If you just update your current `new_branch`, the commits will show up in this PR.

Okay, then let's keep to the 1D case for now, and leave 2D for another PR.

**Comment by taylorpatti** [comment]
I know, but I did this first contribution in an ad hoc way and would like to go forward by pulling CollectiveSpins in developer mode, which pulls from the main master, and then toggle to work on the package, instead of shuffling edited files back and forth. If updating the main is a problem right now though, it's not too big of a deal, I can set it up properly and shuffle files one last time.

**Comment by david-pl** [comment]
Alright, it's merged. Thanks for your work :+1:

**Pull request merged**